src/FixedPointDecimals.jl

```diff
@@ -125,24 +125,36 @@ end
 Base.widemul(x::Integer, y::FD) = widemul(y, x)
 
 """
-    _round_to_even(quotient, remainder, divisor)
-
-Round `quotient + remainder / divisor` to the nearest even integer, given that
-`0 ≤ remainder < divisor` or `0 ≥ remainder > divisor`. (This assumption is
-satisfied by the return value of `fldmod` in all cases, and the return value of
-`divrem` in cases where `divisor` is known to be positive.)
+    _round_to_even(quotient, remainder, divisor, ::RoundingMode{M})
+
+Round `quotient + remainder / divisor` to the nearest even integer,
+given that `0 ≤ remainder < divisor` or `0 ≥ remainder >
+divisor`. (This assumption is satisfied by the return value of
+`fldmod` in all cases, and the return value of `divrem` in cases where
+`divisor` is known to be positive.) The tie is decided depending on
+the `RoundingMode`.
 """
-function _round_to_even(quotient::T, remainder::T, divisor::T) where {T <: Integer}
+function _round_to_even(quotient::T, remainder::T, divisor::T, ::RoundingMode{M}=RoundNearest) where {T <: Integer, M}
     halfdivisor = divisor >> 1
     if iseven(divisor) && remainder == halfdivisor
-        ifelse(iseven(quotient), quotient, quotient + one(quotient))
+        # `:NearestTiesAway` will tie away from zero, e.g. -8.5 ->
+        # -9. `:NearestTiesUp` will always ties towards positive
+        # infinity. `:Nearest` will tie towards the nearest even
+        # integer.
+        if M == :NearestTiesAway && quotient < zero(quotient)
+            quotient
+        elseif M == :Nearest && iseven(quotient)
+            quotient
+        else
+            quotient + one(quotient)
+        end
```

Comment:
I _think_ the `ifelse()` here was probably a perf optimization. We should be careful about changing it.

From my understanding the tradeoff is this:
With `ifelse()`, you always compute both possible values (so you do some "extra" computation) in exchange for avoiding the branch. Since branches are more expensive than the one addition, it's worth it to do the `ifelse`.

((I'm not sure; maybe LLVM would do this anyway though.))

---------

That said, the check on `M` can compile away, so it's just the second parts of those conditions that we need in the ifelse..

I think we can retain the structure of the old code like this:
```suggestion
        if M == :NearestTiesAway
            ifelse(quotient < zero(quotient), quotient, quotient + one(quotient))
        elseif M == :Nearest
            ifelse(iseven(quotient), quotient, quotient + one(quotient))
        else
            quotient + one(quotient)
        end
```